server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodesInfoRequest.java

```diff
@@ -84,174 +85,68 @@ public NodesInfoRequest clear() {
-        return this;
-    }
-
-    /**
-     * Should the node JVM be returned.
-     */
-    public boolean jvm() {
-        return Metrics.JVM.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should the node JVM be returned.
-     */
-    public NodesInfoRequest jvm(boolean jvm) {
-        addOrRemoveMetric(jvm, Metrics.JVM.metricName());
-        return this;
-    }
-
-    /**
-     * Should the node Thread Pool info be returned.
-     */
-    public boolean threadPool() {
-        return Metrics.THREAD_POOL.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should the node Thread Pool info be returned.
-     */
-    public NodesInfoRequest threadPool(boolean threadPool) {
-        addOrRemoveMetric(threadPool, Metrics.THREAD_POOL.metricName());
-        return this;
-    }
-
-    /**
-     * Should the node Transport be returned.
-     */
-    public boolean transport() {
-        return Metrics.TRANSPORT.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should the node Transport be returned.
-     */
-    public NodesInfoRequest transport(boolean transport) {
-        addOrRemoveMetric(transport, Metrics.TRANSPORT.metricName());
-        return this;
-    }
-
-    /**
-     * Should the node HTTP be returned.
-     */
-    public boolean http() {
-        return Metrics.HTTP.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should the node HTTP be returned.
-     */
-    public NodesInfoRequest http(boolean http) {
-        addOrRemoveMetric(http, Metrics.HTTP.metricName());
+    public NodesInfoRequest addMetric(String metric) {
+        if (Metric.allMetrics().contains(metric) == false) {
+            throw new IllegalStateException("Used an illegal metric: " + metric);
+        }
+        requestedMetrics.add(metric);
         return this;
     }
 
     /**
-     * Should information about plugins be returned
-     * @param plugins true if you want info
-     * @return The request
+     * Add collection of metrics
      */
-    public NodesInfoRequest plugins(boolean plugins) {
-        addOrRemoveMetric(plugins, Metrics.PLUGINS.metricName());
+    public NodesInfoRequest addMetrics(Collection<String> metrics) {
+        if (Metric.allMetrics().containsAll(metrics) == false) {
+            throw new IllegalStateException("Used an illegal metric: " + metrics);
+        }
+        requestedMetrics.addAll(metrics);
         return this;
     }
 
     /**
-     * @return true if information about plugins is requested
+     * Add String array of metrics
      */
-    public boolean plugins() {
-        return Metrics.PLUGINS.containedIn(requestedMetrics);
-    }
-
-    /**
-     * Should information about ingest be returned
-     * @param ingest true if you want info
-     */
-    public NodesInfoRequest ingest(boolean ingest) {
-        addOrRemoveMetric(ingest, Metrics.INGEST.metricName());
+    public NodesInfoRequest addMetrics(String... metrics) {
```

Comment:
Why do we need 2 variants of this method?

Reply:
This one made some of the invocations "look cleaner" to me, but it's not at all necessary and I'm fine with removing it.

Reply:
I've removed it.